tools/unix/apply-big-move.sh

```diff
@@ -0,0 +1,181 @@
+  if ! git describe $BEFORE_MOVE_TAG >/dev/null 2>&1; then
+    fail "The Big Move hasn't happened yet (tags not found)."
+  fi
+
+  # Check if we already applied The Big Move in this branch, as indicated by $AFTER_MOVE_TAG being
+  # in our history.
+  if git merge-base --is-ancestor $AFTER_MOVE_TAG HEAD; then
+    # This branch already includes The Big Move, but maybe there is a future Big Move and the
+    # user was actually intending to apply that?
+    if git describe $NEXT_BIG_MOVE >/dev/null 2>&1; then
+      # Indeed there is, let's skip forward to that.
+      rebase_and_rerun $NEXT_BIG_MOVE
+    fi
+
+    echo "The Big Move has already been applied to this branch."
+    exit 0
+  fi
+
+  # Check if $BEFORE_MOVE_TAG is in this branch's history. If not, we need to rebase first.
+  if ! git merge-base --is-ancestor $BEFORE_MOVE_TAG HEAD; then
+    # Branch is not yet based on $BEFORE_MOVE_TAG, so rebase onto it.
+    rebase_and_rerun $BEFORE_MOVE_TAG
+    fail "(can't get here -- rebase_and_rerun should not return)"
+  fi
+
+  # Get the list of commits that we need to rebase over The Big Move.
+  COMMITS=($(git log --reverse --format='%H' $BEFORE_MOVE_TAG..HEAD))
+
+  # Checkout $AFTER_MOVE_TAG in order to start building on it.
+  git checkout -q $AFTER_MOVE_TAG
+  git submodule update
+
+  # Apply each commit.
+  for COMMIT in ${COMMITS[@]}; do
+    git log -1 --format='%CblueRewriting commit %h:%Creset %s' $COMMIT
+
+    # Update the working tree to match the code from the source commit.
+    git checkout $COMMIT .
+    git submodule update
+
+    # Apply the Big Move on top of that.
+    apply_big_move
+
+    # Commit to edgeworker (including updating the workerd submodule), reusing the old commit
+    # message. It's possible that the diff is now empty, in which case we skip this commit, much
+    # like `git rebase` does by default.
+    if git commit -a --dry-run > /dev/null 2>&1; then
+      git commit -aC $COMMIT
+    fi
+  done
+
+  # Record the final commit.
+  FINAL_COMMIT=$(git log -1 --format=%H)
+
+  # Check out the original branch, and reset it to the final commit.
+  git checkout -q "$ORIG_BRANCH"
+  git reset --hard "$FINAL_COMMIT"
+
+  # Success!
+  echo -e '\e[0;1;32mSUCCESS:\e[0m Applied Big Move.'
+
+  # Check if there's another move upcoming that we should apply next.
+  check_for_next_move
+
+  exit 0
+}
+
+# Ensure that if the file changes while bash is reading it, bash won't go off the rails. Normally,
+# bash reads one line from the file at a time and then executes it before reading the next.
+# Wrapping the whole script body in a function ensures that bash buffers it in memory. Placing
+# `exit 0` on the same line ensures that bash won't attempt to read anything more after the
+# function returns.
+#
+# This ensures that if further changes are made to this file before the Big Move actually happens,
+# then the rebase command that rebases to the commit before the Big Move won't confuse bash.
+main "$@"; exit 0
```

Comment:
I skimmed over this, assuming that everything below `apply_big_move()` is unchanged. LMK if not.